tests/printer/printer.cpp

```diff
@@ -295,3 +293,257 @@ TEST(Printer, printModelWithStandardUnitsAdded)
+                           "    <reset variable=\"variable1\" test_variable=\"variable2\" order=\"1\">\n"
+                           "      <test_value id=\"test_value_1\">\n"
+                           "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\">\n"
+                           "          <apply>\n"
+                           "            <eq/>\n"
+                           "            <ci>variable1</ci>\n"
+                           "            <cn cellml:units=\"units2\">3.4</cn>\n"
+                           "          </apply>\n"
+                           "        </math>\n"
+                           "      </test_value>\n"
+                           "      <reset_value>\n"
+                           "        <math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\">\n"
+                           "          <apply>\n"
+                           "            <eq/>\n"
+                           "            <ci>variable1</ci>\n"
+                           "            <cn cellml:units=\"units2\">9.0</cn>\n"
+                           "          </apply>\n"
+                           "        </math>\n"
+                           "      </reset_value>\n"
+                           "    </reset>\n"
+                           "  </component>\n"
+                           "  <component name=\"component3\">\n"
+                           "    <variable name=\"variable4\" units=\"units2\" interface=\"public\"/>\n"
+                           "    <variable name=\"variable2\" units=\"units2\" interface=\"public\"/>\n"
+                           "    <math xmlns=\"http://www.w3.org/1998/Math/MathML\" xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\">\n"
+                           "      <apply>\n"
+                           "        <eq/>\n"
+                           "        <ci>variable4</ci>\n"
+                           "        <cn cellml:units=\"units2\">9.0</cn>\n"
+                           "      </apply>\n"
+                           "    </math>\n"
+                           "  </component>\n"
+                           "  <connection component_1=\"component2\" component_2=\"component3\">\n"
+                           "    <map_variables variable_1=\"variable1\" variable_2=\"variable2\"/>\n"
+                           "    <map_variables variable_1=\"variable1\" variable_2=\"variable4\"/>\n"
+                           "  </connection>\n"
+                           "  <encapsulation>\n"
+                           "    <component_ref component=\"component2\">\n"
+                           "      <component_ref component=\"component3\"/>\n"
+                           "    </component_ref>\n"
+                           "  </encapsulation>\n"
+                           "</model>\n";
+    const std::string expectedOutString = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```

Comment:
what happened to the random/hexadecimal IDs?

Reply:
They are in a different PR, see https://github.com/cellml/libcellml/pull/639